### PR TITLE
feat: support N nginx ingresses

### PR DESCRIPTION
**Pull request by tperdue321**
Enhancement proposal: https://github.com/argoproj/argo-rollouts/issues/2001

## Modifications:

- update the `Nginx` datatype to have a field called `AdditionalStableIngresses` to allow for 1-N stable ingresses instead of just 1. This naming convention is modeled after `AdditionalIngressAnnotations`
- modify the `SetWeight` method to create a canary for each stable ingress and apply new weight to all canary ingresses.
- update validations to validate all stable ingresses.
- designed to be fully backwards compatible.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by ssanders1449** [comment]
@zachaller - Is this targeted for 1.4?

**Comment by zachaller** [comment]
@ssanders1449 also note I do want to follow a much more open release schedule here is the proposal https://github.com/argoproj/argo-rollouts/pull/2446/files

**Comment by tperdue321** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=argoproj_argo-rollouts&pullRequest=2467)
> 
> [Bug](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=CODE_SMELL) [36 Code Smells](https://sonarcloud.io/project/issues?id=argoproj_argo-rollouts&pullRequest=2467&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=argoproj_argo-rollouts&pullRequest=2467) No Coverage information [](https://sonarcloud.io/component_measures?id=argoproj_argo-rollouts&pullRequest=2467&metric=new_duplicated_lines_density&view=list) [43.6% Duplication](https://sonarcloud.io/component_measures?id=argoproj_argo-rollouts&pullRequest=2467&metric=new_duplicated_lines_density&view=list)

I'm working on reducing duplication today

**Comment by zachaller** [comment]
@tperdue321 Can I get you to resovle conflicts before I review?

**Comment by zachaller** [comment]
Also can you update from master to resolve conflicts.

File: ingress/ingress_test.go [hunk 1]
```diff
@@ -24,6 +26,16 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
+const StableService string = "test-stable-service"
+const AdditionalStableService string = "test-stable-service-additional"
```
Comment (zachaller) [inline ingress/ingress_test.go:30]: Is this used?

File: ingress/ingress_test.go [hunk 2]
```diff
@@ -24,6 +26,16 @@ import (
...
+const AdditionalIngress string = "test-stable-ingress-additional"
```
[the remark this answers is not in the document]
Comment (tperdue321) [reply ingress/ingress_test.go:31]: yep good catch thanks

File: ingress/ingress_test.go (hunk 1 above)
Comment (tperdue321) [reply ingress/ingress_test.go:30]: no, I believe this was missed cleanup when I was changing tests based off of the change to have `stableIngress`/`stableIngresses` be exclusive fields. thanks!

**Review by zachaller — APPROVED** [review]

**Pull request merged**